src/scm/bitbucket.js

```diff
@@ -1,4 +1,305 @@
+      }
+      $.ajax({
+        url: `${this.baseUrl}/repositories/${repo}/src`,
+        headers: {
+          'Authorization': `Bearer ${this.accessToken}` 
+        },
+        contentType: 'application/x-www-form-urlencoded',
+        method: 'POST',
+        crossDomain: true,
+        traditional: true,
+        data: data,
+      })
+      .then(resolve)
+      .fail(reject);
+    });
+  }
+
+  push(code){
+    const changed = $('.diff-file:checked').toArray().map(elem => elem.value);
+    const files = changed.filter(f => code.gas[f]).map(f => {
+      return { name: f, content: code.gas[f] }
+    });
+    const deleteFiles = changed.filter(f => !code.gas[f]);
+    const comment = $('#commit-comment').val();
+
+    this.commitFiles(context.repo.fullName, context.branch, null, files, deleteFiles, comment)
+    .then(() => {
+      showAlert(`Successfully push to ${context.branch} of ${context.repo.fullName}`);
+    })
+    .catch((err) => {
+      showAlert('Failed to push', LEVEL_ERROR);
+    });
+  }
+
+  getAllBranches() {
+    return this.getAccessToken()
+    .then(accessToken => {
+      return getAllItems(Promise.resolve(
+        {
+          token: accessToken,
+          items: [], 
+          url: `${this.baseUrl}/repositories/${context.repo.fullName}/refs/branches?access_token=${accessToken}`
+        }),
+        this.followPaginate,
+        'bitbucket'
+      );
+    });
+  }
+
+  getCode() {
+    return this.getAccessToken()
+    .then(accessToken => {
+      return $.getJSON(
+        `${this.baseUrl}/repositories/${context.repo.fullName}/refs/branches/${context.branch}`,
+        { access_token: accessToken }
+      )
+    })
+    .then(response => {
+      return getAllItems(Promise.resolve(
+        {
+          token: this.accessToken,
+          items: [], 
+          urls: [],
+          url: `${this.baseUrl}/repositories/${context.repo.fullName}/src/${response.target.hash}/?access_token=${this.accessToken}`
+        }),
+        this.followDirectory,
+        'bitbucket'
+      )
+      .then(response => {
+        const promises = response.map(src => {
+          return new Promise((resolve, reject) => {
+            $.get(src.links.self.href, { access_token: this.accessToken })
+            .then(content => {
+              resolve({ file: src.path, content: content});
+            })
+            .fail(reject)
+          });
+        });
+        return Promise.all(promises);
+      });
+    });
+  }
+
+  getNamespaces() {
+    return this.getAccessToken()
+    .then(accessToken => {
+      return getAllItems(Promise.resolve(
+        {
+          token: accessToken,
+          items: [], 
+          url: `${this.baseUrl}/teams?access_token=${accessToken}&role=contributor`
+        }),
+        this.followPaginate,
+        'bitbucket'
+      );
+    })
+    .then(teams => {
+      this.namespaces = [this.user].concat(teams.map(team => team.username));
+      return this.namespaces;
+    })
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/79a1121575f894385b70c00ba30d867da9fde2de/src/scm/bitbucket.js#L163"></span>‘err’ is defined but never used.</td>
    </tr>
  </tbody>
</table>